src/main/java/net/javadiscord/javabot/listener/PingableNameListener.java

```diff
@@ -0,0 +1,115 @@
+package net.javadiscord.javabot.listener;
+
+import lombok.extern.slf4j.Slf4j;
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.events.guild.member.GuildMemberJoinEvent;
+import net.dv8tion.jda.api.events.guild.member.update.GuildMemberUpdateNicknameEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import net.javadiscord.javabot.util.StringUtils;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Checks User's names & nicknames and makes sure they're pingable.
+ */
+@Slf4j
+public class PingableNameListener extends ListenerAdapter {
+
+	private static final String ADJECTIVES_URL = "https://gist.githubusercontent.com/karlbright/f91229b8c5ac6f4291dc/raw/4a69c2c50b88ee4559b021c443fee899535adc60/adjectives.txt";
+	private static final String NOUNS_URL = "https://raw.githubusercontent.com/hugsy/stuff/main/random-word/english-nouns.txt";
+	private static final Random random = new Random();
+	private final List<String> nouns;
+	private final List<String> adjectives;
+
+	/**
+	 * Constructs a new PingableNameListener and loads nouns & adjectives.
+	 */
+	public PingableNameListener() {
+		nouns = readStrings(NOUNS_URL);
+		adjectives = readStrings(ADJECTIVES_URL);
+		log.info("Loaded {} Nouns!", nouns.size());
+		log.info("Loaded {} Adjectives!", adjectives.size());
+	}
+
+	@Override
+	public void onGuildMemberJoin(GuildMemberJoinEvent event) {
+		if (!isPingable(event.getUser().getName()) && !canBypassCheck(event.getMember())) {
+			String newName = generateRandomName();
+			event.getMember().modifyNickname(newName).queue();
+			event.getUser().openPrivateChannel()
+					.flatMap(channel -> channel.sendMessageFormat("Your nickname has been set to %s since your username's first three characters were deemed as not-pingable.", newName))
+					.queue();
+		}
+	}
+
+	@Override
+	public void onGuildMemberUpdateNickname(GuildMemberUpdateNicknameEvent event) {
+		if (!isPingable(event.getNewNickname()) && !isPingable(event.getUser().getName()) && !canBypassCheck(event.getMember())) {
+			String newName = generateRandomName();
+			event.getMember().modifyNickname(newName).queue();
+			event.getUser().openPrivateChannel()
+					.flatMap(channel -> channel.sendMessageFormat("Your nickname has been set to `%s` since both your user- and nickname's first three characters were deemed as not-pingable.", newName))
+					.queue();
+		}
+	}
+
+	/**
+	 * Checks if the given name's first three characters contain ASCII characters not between 32 and 126.
+	 * @param name The name to check.
+	 * @return True if first three characters contain invalid characters, False if not.
+	 */
+	private boolean isPingable(String name) {
+		if (name == null) return true;
+		char[] nameChars = name.toCharArray();
+		for (int i = 0; i < 2; i++) {
```

Comment:
This throws a NPE if the name has less than 3 characters.
```suggestion
		for (int i = 0; i < Math.min(2,name.length()); i++) {
```